### PR TITLE
Fixing mad click in sample miner script

### PR DESCRIPTION
**Pull request by philtreF**
Hi, 

I added some code in `DefenseStep` in order to count the number of drones which are currently fighting. 

This makes me able to prevent miner bot to mad click "Engage NPC" :  It is needed  to click on "Engage target" only when the number of drones in space is not equal with the number of drone which are currently fighting. 

This is my first contribution to this project, do not hesitate to give me feedback on ways to improve for next contributions. 

rh-fred.

### CODE REVIEW

**Comment by Viir** [comment]
Thank you for the changes to the mining script.

I do not understand why the changes are divided into two commits.
The message from c55e52e ready like it only changes whitespaces.
Why not squash c8bc277 and c55e52e into one commit?

File: src/Sanderling/Sanderling.Exe/sample/script/Mine.ore.cs [hunk 1]
```diff
@@ -100,8 +100,10 @@
 long? JammedLastAge => Jammed ? 0 : (Host.GetTimeContinuousMilli() - JammedLastTime) / 1000;
 
 int?	ShieldHpPercent => ShipUi?.HitpointsAndEnergy?.Shield / 10;
+int droneFightingCounter = 0;
```
Comment (Viir) [inline src/Sanderling/Sanderling.Exe/sample/script/Mine.ore.cs]: Why is `droneFightingCounter ` scoped to the whole script?
What is a case where its state is actually reused between method calls?

**Comment by fhilpert** [comment]
About the variable scope, I used to print the value in main `Host.Log` that is why it is there. 

I will correct all theses, merge commits (didn"t know this is possible) and tell you when it will be done.

**Comment by tetreum** [comment]
@fhilpert any progress?

**Pull request closed without merging**